src/kiota/Handlers/KiotaInfoCommandHandler.cs

```diff
@@ -112,15 +117,23 @@ private void ShowLanguagesTable()
         var layout = new StackLayoutView { view };
         console.Append(layout);
     }
-    private void ShowLanguageInformation(GenerationLanguage language, LanguagesInformation informationSource)
+    private void ShowLanguageInformation(GenerationLanguage language, LanguagesInformation informationSource, bool json)
     {
         if (informationSource.TryGetValue(language.ToString(), out var languageInformation))
         {
-            DisplayInfo($"The language {language} is currently in {languageInformation.MaturityLevel} maturity level.",
-                        "After generating code for this language, you need to install the following packages:");
-            foreach (var dependency in languageInformation.Dependencies)
+            if (!json)
+            {
+                DisplayInfo($"The language {language} is currently in {languageInformation.MaturityLevel} maturity level.",
+                            "After generating code for this language, you need to install the following packages:");
+                foreach (var dependency in languageInformation.Dependencies)
+                {
+                    DisplayInfo(string.Format(dependency.Name, dependency.Version));
+                }
+            }
+            else
             {
-                DisplayInfo(string.Format(languageInformation.DependencyInstallCommand, dependency.Name, dependency.Version));
+                var jsonOutput = "[" + string.Join(",", languageInformation.Dependencies.Select(d => "{\"" + d.Name + "\":\"" + d.Version + "\"}")) + "]";
```

Comment:
I'm a bit worried about this approach. We're effectively creating a "public API" we have to maintain compatibility with.
in #3019 we've been discussing about adding categories for example.
I'd much rather have a root "DependenciesExport" object that has a "dependencies" list. This way if we want to add a list of categories (their descriptions) we'll have more flexibility.
And use System.Text.Json to handle the serialization to reduce the future maintenance burden.
Thoughts?

Reply:
Agree!
Checked and `LanguageInformation` is already "public API" of the `Kiota.Builder` package, all of the compatibility reasoning should already apply as far as I understand it.

Used it directly to emit the output, wdyt?